Redesign UI to Match Proposed Design.

Glorycornel opened the pull request:
### ✅ **What’s Changed**

This PR introduces a full UI revamp based on the latest design proposal:

#### 🔧 **UI Updates**
- Redesigned the quiz generation form to match the new layout and spacing
- Updated the styling of buttons (`Generate`, etc.) for better visual hierarchy and interaction feedback

#### 🧩 **New Components**
- Created new reusable UI components:

#### 🧭 **Navigation**
- Added a new navbar component with styled branding/navigation


### 🧪 **How to Test**

1. **Pull this branch locally**
2. Run the dev server:
   ```bash
   npm run dev
   ```
3. Navigate to the quiz generation page
4. Confirm that:
   - The layout matches the new design
   - Form is responsive on mobile and desktop
   - Quiz can be generated successfully via the "Generate Quiz" button
   - Navigation bar is visible and styled correctly

### 📌 **Notes**
- This PR only covers UI/UX — functionality is unchanged.

### 📦 **Dependencies**

Before testing, install the required dependencies:

```bash
pnpm install axios
pnpm add -D tailwindcss postcss autoprefixer
pnpm dlx tailwindcss init -p
```

![Screenshot from 2025-05-02 11-54-26](https://github.com/user-attachments/assets/dbd8d4cc-6c0e-453d-9509-18d73694d751)
![Screenshot from 2025-05-02 11-54-55](https://github.com/user-attachments/assets/acccd332-179d-4541-b9a3-18578efc9a9c)

fabricewin reviewed (approved):
![image](https://github.com/user-attachments/assets/59799977-51ed-4ad7-ab44-2f79614d1432)

Glorycornel commented:
Replaced fetch with Axios on the quiz form, and also added the sidebar to the nav 
![Screenshot from 2025-05-02 11-54-33](https://github.com/user-attachments/assets/7918cf54-4e67-42b1-888d-7ec94da190c8)
![Screenshot from 2025-05-02 11-54-26](https://github.com/user-attachments/assets/f84501c2-6e31-4bb4-aded-f14a9a87b2d2)
